### PR TITLE
MVP

### PR DESCRIPTION
**Pull request by ecwillis**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+Attribute | Description
+--- | --- | ---
+`status` | An enumeration of the request status. `ok` for a successful transaction, and `error` if there was an error. The HTTP status code will also reflect this status (`200` for `ok`, `4**`, `5**` for error)
+`message` | A Pretty message from the system.
+`result` | A proxy result from the mailing system. I.E. response from the Mandrill API.
+
+-----
+
+`POST /cache`
```
Comment (jgallen23) [inline README.md]: we don't need this anymore

File: conf/default.yaml [hunk 2]
```diff
@@ -1,19 +1,17 @@
...
 plugins:
   vision:
   inert:
-  ./plugins/mailer:
-    smtp:
-      host: "{{ENV.SMTP_HOST}}"
-      port: "{{ENV.SMTP_PORT}}"
-      user: "{{ENV.SMTP_USER}}"
-      pass: "{{ENV.SMTP_PASS}}"
+  ./plugins/mandrill:
```
Comment (jgallen23) [inline conf/default.yaml]: why did this change?

File: docker-compose.yml [hunk 3]
```diff
@@ -7,3 +7,5 @@ app:
     PORT: 8081
   ports:
     - '8081:8081'
+  external_links:
+    - 'mongo:mongo'
```
Comment (jgallen23) [inline docker-compose.yml]: what do we need mongo for?

File: package.json [hunk 4]
```diff
@@ -25,18 +25,22 @@
   "homepage": "https://github.com/firstandthird/micro-mail#readme",
   "dependencies": {
     "async": "2.0.0-rc.4",
-    "rapptor": "firstandthird/rapptor#e1b04d6",
+    "handlebars": "4.0.5",
+    "inert": "4.0.0",
+    "mandrill-api": "^1.0.45",
```
Comment (jgallen23) [inline package.json]: why are we loading mandrill-api?

**Comment by jgallen23** [comment]
closing

**Comment by ecwillis** [comment]
Reopening post fixes.

File: plugins/mailer/send.js [hunk 5]
```diff
@@ -1,19 +1,68 @@
...
+const yamljs = require('yamljs');
+const handlebars = require('handlebars');
+const _ = require('lodash');
+
+
 const async = require('async');
 module.exports = function(server, transporter, emailData, allDone) {
   async.auto({
+    details: (done) => {
+      const template = emailData.template || null;
```
Comment (jgallen23) [inline plugins/mailer/send.js]: we can just use that method you created here

File: routes/send.js [hunk 6]
```diff
@@ -2,8 +2,24 @@ exports.send = {
   path: '/send',
   method: 'POST',
   handler(request, reply) {
-    const emailData = request.payload;
-    request.server.sendEmail(emailData, reply);
+    // Render emailDetails
+    const renderedData = request.server.methods.renderData(request.payload);
```
Comment (jgallen23) [inline routes/send.js]: is this running the render stuff twice?

File: routes/view.js [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+const _ = require('lodash');
+
+exports.view = {
+  path: '/view/{email}',
+  method: 'GET',
+  handler(request, reply) {
+    const email = request.params.email;
+    let defaultData = {};
+    try {
+      defaultData = require(`${request.server.settings.app.templatePath}/${email}/debug`);
```
Comment (jgallen23) [inline routes/view.js]: debug -> test

**Pull request merged**